### PR TITLE
[TR] PIM-3980: Proposals tab

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-3980

| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | -
| Specs & Static     | -
| Changelog updated  | N

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js [hunk 1]
```diff
@@ -51,9 +55,14 @@ define(
...
 
+                var state = this.state.toJSON();
+                state.tabs = _.filter(state.tabs, function (tab) {
+                    return !_.isFunction(tab.displayCondition) || tab.displayCondition();
+                });
+
                 this.$el.html(
                     this.template({
-                        state: this.state.toJSON()
+                        state: state
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js]: what if the current tab is not in the filtered tabs ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js [hunk 2]
```diff
@@ -36,7 +36,11 @@ define(
             },
             registerTab: function (event) {
                 var tabs = this.state.get('tabs') || [];
-                tabs.push({ code: event.code, label: event.label });
+                tabs.push({
+                    code: event.code,
+                    displayCondition: event.displayCondition,
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js]: what about `isVisible` ?

Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js]: Good idea

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/form-tabs.js]: If it has been filtered it means that you're not supposed to see it. So if it's the current tab there's a problem.

**Comment by grena** [comment]
GTM :+1:

**Pull request merged**